back/boxtribute_server/business_logic/box_transfer/shipment/crud.py

```diff
@@ -146,6 +146,22 @@ def send_shipment(*, id, user):
     return shipment
 
 
+def receive_shipment(*, id, user):
+    """Transition state of specified shipment to 'Receiving'.
+    Raise InvalidShipmentState exception if shipment state is different from 'Sent'.
+    """
+    shipment = Shipment.get_by_id(id)
+    if shipment.state != ShipmentState.Sent:
+        raise InvalidShipmentState(
+            expected_states=[ShipmentState.Sent], actual_state=shipment.state
+        )
+    shipment.state = ShipmentState.Receiving
+    # shipment.received_by = user.id
```

Comment:
what happened here @pylipp ? Is this about clarifying dates with @aerinsol ?

Reply:
needs a database migration first @HaGuesto 
Then shipment table should look like:
```
  `received_on` datetime DEFAULT NULL,
  `received_by_id` int unsigned DEFAULT NULL,
  CONSTRAINT `shipment_ibfk_7` FOREIGN KEY (`received_by_id`) REFERENCES `cms_users` (`id`) ON DELETE SET NULL ON UPDATE CASCADE
```

Reply:
https://trello.com/c/omJaOa2A/1131-add-receivedon-and-receivedby-fields-to-shipments-db-table